editor/editor_help_search.cpp

```diff
@@ -590,19 +657,49 @@ TreeItem *EditorHelpSearch::Runner::_create_class_hierarchy(const ClassMatch &p_
 	return class_item;
 }
 
+TreeItem *EditorHelpSearch::Runner::_find_or_create_item(TreeItem *p_parent, const String &p_item_meta) {
+	TreeItem *item = nullptr;
+
+	// Attempt to find in cache.
+	if (tree_cache->item_cache.has(p_item_meta)) {
+		item = tree_cache->item_cache[p_item_meta];
+
+		// Remove from cache.
+		tree_cache->item_cache.erase(p_item_meta);
+
+		// Add to tree.
+		p_parent->add_child(item);
+
+		// Erase buttons if any, to update icons.
+		for (int i = 0; i < item->get_button_count(0); ++i) {
+			item->erase_button(0, i);
+		}
+	} else {
+		// Otherwise create item.
+		item = results_tree->create_item(p_parent);
+	}
+
+	return item;
+}
+
 TreeItem *EditorHelpSearch::Runner::_create_class_item(TreeItem *p_parent, const DocData::ClassDoc *p_doc, bool p_gray) {
 	String tooltip = DTR(p_doc->brief_description.strip_edges());
 
-	TreeItem *item = results_tree->create_item(p_parent);
+	const String item_meta = "class_name:" + p_doc->name;
+
+	TreeItem *item = _find_or_create_item(p_parent, item_meta);
 	item->set_icon(0, EditorNode::get_singleton()->get_class_icon(p_doc->name));
 	item->set_text(0, p_doc->name);
```

Comment:
I think `_find_or_create_item` could return a boolean instead and populate the item passed by reference. This way we would only set all these values once, initially. I don't see a reason to do it every time. We can make the exception for `p_gray` which is passed as an argument right now (although it's not something that's supposed to be mutable, I don't think, and should be more explicitly stored in the class reference cache).

And I'm not sure about the buttons. If we need to refresh them somehow, but don't have API for that, we could add it. But neither `is_deprecated` nor `is_experimental` can change between searches. It can only change in scripted classes when you change their source, and we clear the cache anyway when the dialog disappears.

Reply:
Will look into it! Should work well

Reply:
Was trivial, done!